__tests__/createCostume.test.js

```diff
@@ -0,0 +1,37 @@
+const {client, createTables, createCostume} = require(".././index");
+
+describe("createCostume adapter", () => {
+    // Connect to postgres database and create table before tests
+    beforeAll(async () => {
+        await client.connect();
+        console.log("connected");
+        await createTables();
+    })
+    // Disconnect from postgres database after tests
+    afterAll(async () => {
+        await client.end();
+        console.log("connection closed");
+    })
+
+    it("should create a new row in the table", async () => {
```

Comment:
You may also want to check other test cases, things not on the "happy path" - eg. it throws an error if not given enough arguments, or maybe test your DB constraints - throws an error if certain invalid items are passed for example

Reply:
I thought about that, but didn't know if I was on the right track as far as catching the error with Jest. Would it be adding `.toThrow()` to the adapter call? For example, `await createCostume(...).toThrow()`?

Reply:
Yes, that would be appropriate - and I would recommend including the argument for `.toThrow()` so that you can confirm that you are seeing exactly the error you expect.

Reply:
I'm attempting to write a test that expects an error to be thrown if `createCostume` has too few arguments. The error message that it logs when I do that is: `error: null value in column "price" of relation "costumes" violates not-null constraint`. But when I wrote the code below, the test failed and Jest said that `received function did not throw`. I take it `createCostume` did not throw the error I had seen in the console? What function did? If I knew that, I would wrap it in the `expect` call to try that out. I can't figure out what threw the error. I have been searching for a way to find that out in this scenario, but coming up dry on Google. Here is the stack trace when the error is thrown: 
```
      at node_modules/pg-pool/index.js:45:11
      at createCostume (index.js:66:30)
      at Object.<anonymous> (__tests__/costumes.test.js:133:9)
```

The test that is failing:
```
it.only("should throw an error if not given enough arguments", async () => {
        await createTables(pool);
        expect(async () => {
            await createCostume(
                pool,
                "mutton chops",
                "adult",
                "male",
                "M",
                "facial hair",
                6
            )
        }).toThrow(new Error(`error: null value in column "price" of relation "costumes" violates not-null constraint`))
    })

Reply:
I've been looking at ways to edit the adapter so that `createCostume` throws its own error if there aren't enough arguments. I was trying to use the arguments object and use `arguments.length`, but that is telling me the length is 6 every time no matter how many arguments are passed. I know the arguments object is a little outdated as far as use, but that was the best method. This may all be moot because I think if i don't have enough arguments, Postgres throws the error because a not null constraint has not been met, and I still haven't found a way to test that error with `toThrow()` or anything else.